### DISCUSSION TITLE
Can homespan use with iphone from Internet?

### DISCUSSION THREAD

**Discussion by Buffer2000** [HomeSpan/HomeSpan#148] [Q&A] [answered]
I try homespan with my ESP32 pairing and work flawlessly. However, when I try disconnect Wifi on iphone, I cannot contact to the homespan device on ESP32.

Are the homespan designed to working with the same Wifi only or there is a way to make it accessible from Internet? And if yes, please hint me how to or even where to find info.

**Comment by HomeSpan** [comment, marked as answer]
HomeSpan is designed to work when (and only when) connected to your home's WiFi network.  Before any pairing can occur you need to make sure you've given HomeSpan you network's WiFi Credentials.  The easiest way to do this is by using the 'W' command in the Serial Monitor.   Alternatively, you can launch HomeSpan's temporary WiFi Access Point, which presents a web page into which you can type your WiFi Credentials.  This is only needed if you have to configure a HomeSpan device that is not connected to a computer with a USB.

**Reply by Buffer2000** [reply]
Thank you for clearing that.
I also search and found some info that to use iphone accessing homekit/homespan from Internet we need a thing call home hub which now officially can be only ipad or apple TV.

**Reply by gpena20877** [reply]
> HomeSpan is designed to work when (and only when) connected to your home's WiFi network. Before any pairing can occur you need to make sure you've given HomeSpan you network's WiFi Credentials. The easiest way to do this is by using the 'W' command in the Serial Monitor. Alternatively, you can launch HomeSpan's temporary WiFi Access Point, which presents a web page into which you can type your WiFi Credentials. This is only needed if you have to configure a HomeSpan device that is not connected to a computer with a USB.

Documentation mentions that if no wifi credientials are found, the temp AP is created automatically. However I do not see it under wifi networks on the iPhone.

I also have this code in my setup portion, above the homespan.begin

`void setApTimeout(uint16_t nSec);
  void enableAutoStartAP();`
But it still doesnt generate the temp AP when the device is reset and powered on.

**Reply by KhazanGu** [reply]
> > HomeSpan is designed to work when (and only when) connected to your home's WiFi network. Before any pairing can occur you need to make sure you've given HomeSpan you network's WiFi Credentials. The easiest way to do this is by using the 'W' command in the Serial Monitor. Alternatively, you can launch HomeSpan's temporary WiFi Access Point, which presents a web page into which you can type your WiFi Credentials. This is only needed if you have to configure a HomeSpan device that is not connected to a computer with a USB.
> 
> Documentation mentions that if no wifi credientials are found, the temp AP is created automatically. However I do not see it under wifi networks on the iPhone.
> 
> I also have this code in my setup portion, above the homespan.begin
> 
> `void setApTimeout(uint16_t nSec); void enableAutoStartAP();` But it still doesnt generate the temp AP when the device is reset and powered on.

You could use [CLI](https://github.com/HomeSpan/HomeSpan/blob/master/docs/CLI.md) or [Control button](https://github.com/HomeSpan/HomeSpan/blob/master/docs/GettingStarted.md#adding-a-control-button-and-status-led-optional) to make the device enter Device Configuration Mode and after that, you need to choose mode 3 If you would like to start an AP.

**Reply by KhazanGu** [reply]
> Thank you for clearing that. I also search and found some info that to use iphone accessing homekit/homespan from Internet we need a thing call home hub which now officially can be only ipad or apple TV.

From Apple:
> You need a home hub to control your HomeKit accessories while you're away from your home, grant access to the people you trust, and automate your accessories to do what you want, when you want.

For your reference:
https://support.apple.com/en-us/HT207057

**Comment by HomeSpan** [comment]
To enable the AP to autostart you need to _call_ the method enableAutoStartAP().  In C++ the commands you listed above _declare_ new functions setAPTimeout() and enableAutoStartAP(), which is likely not what you want to do.  Note that HomeSpan documentation follows standard C and C++ conventions in which the functions declarations are presented.  But that's not how you call the function in your program.

To call the enableAutoStartAP(), you would use `homeSpan.enableAutoStartAP();`. This will cause HomeSpan to autostart the AP if WiFi credentials are not found.

**Reply by gpena20877** [reply]
Thanks for the fast reply. That worked nicely. Oddly enough, the name broadcasted is `ESP_B0921D`
